tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,11 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+
+			ClassLoader cl = NullnessUtil.castNonNull( this.getClass().getClassLoader() );
```

Comment:
OK, that's a lot more understandable now, but I still don't get it.

Why would we not use `ObjectFactory.class.getClassLoader()` if we're trying to load things in that package?

Reply:
That's also a mystery to me. Why would `ObjectFactory.class.getClassLoader()` be different from `this.getClass().getClassLoader()`?

Reply:
I can change this for sure if you think it'll be more readable. The goal is only to get the current classloader

Reply:
Yes, please change it and make sure that it still solves the problem.

Reply:
I'll do that in the next hour for now 😀

Reply:
Many thanks for the review at both of you 🤗

Reply:
@gavinking and @beikov : still working with latest change (`ObjectFactory.class.getClassLoader()` instead of `this.getClass().getClassLoader()`). Just forced push this change now